plugin/delimitMate.vim

```diff
@@ -368,6 +368,11 @@ augroup delimitMate
 				\   call <SID>DelimitMateDo() |
 				\   let b:delimitMate_was_here = 1 |
 				\ endif
+
+	" Keep track of visualmode(), which is used in s:get_left_motion(),
+	" to skip leaving insert mode in block-wise visual mode.
+	autocmd InsertEnter * let b:delimitMate_visualmode = visualmode()
```

Comment:
Yes, but you could track the last known mode with a cursormoved autocmd (and maybe a few more like I did) to get the current mode as the cursor moves. Store that in a variable, and use that VARIABLE instead of visualmode() on InsertEnter. That's what my suggestion does. I think you could still "trick" it in many ways so it's not a good final solution, however. I think mapping every possible insert key from visual mode may sadly be the answer.